docs/content/tutorial-cmake.md

```diff
@@ -0,0 +1,146 @@
+CMake Tutorial {#tut_cmake}
+====
+\brief Generating LCM type bindings with CMake
+
+This tutorial will walk you through a sample `CMakeLists.txt` that will
+generate bindings for your LCM message types.
+
+First, of course, you will want to find LCM itself, and include it's "use file"
+(a "use file" is a CMake script provided with some packages that adds utility
+functions for using that package). This can be done in your root
+`CMakeLists.txt`. If you will be generating Python and/or Java bindings, now
+is a good time to find those also.
+
+    find_package(lcm REQUIRED)
+    include(${LCM_USE_FILE})
+
+    find_package(PythonInterp)
+    find_package(Java)
+
+    if(JAVA_FOUND)
+      include(UseJava)
+    endif()
+
+Add `REQUIRED` to the `find_package` calls if you want to enforce that Python
+and/or Java are available. Otherwise, you can generate bindings for these
+languages conditionally. (The rest of the logic we show, including the next
+snippet, will go in the `CMakeLists.txt` that's located alongside your LCM
+types.)
+
+    if(PYTHONINTERP_FOUND)
+      set(python_args PYTHON_SOURCES python_install_sources)
+    endif()
+    if(JAVA_FOUND)
+      set(java_args JAVA_SOURCES java_sources)
+    endif()
+
+(The main reason to find Python is to obtain its version so that the generated
+Python files can be installed into the correct directory. If you determine this
+directory by some other means, you can skip finding Python and just generate
+the Python bindings unconditionally.)
+
+Now, you'll want to generate the bindings.
+
+    lcm_wrap_types(
+      C_SOURCES sources
+      C_HEADERS c_install_headers
+      CPP_HEADERS cpp_install_headers
+      ${python_args}
+      ${java_args}
+      my_type_1.lcm
+      my_type_2.lcm
+      ...
+    )
+
+This will create the necessary rules to run `lcm-gen`, and will set several
+CMake variables that you will use later.
+
+This is really all you need to know that is specific to LCM, but for the sake
+of completeness, we'll show how the rest of the process of creating stand-alone
+bindings libraries (that is, libraries that contain no code besides the LCM
+type bindings themselves). Next, we'll tell CMake to build a library and link
```

Comment:
At least the C and C++ portions are intertwined. They *could* be separated, but it would cause unnecessary duplication of logic. The remaining Python bit is a single line guarded by an `if(PYTHON_FOUND)`. And all of them are already intertwined in the preceding logic.